crawler/file_processing.py

```diff
@@ -1138,48 +1157,48 @@ def row_channel_result_values_valid(self, row, line_number) -> bool:
 
         return True
 
-    def is_within_cq_range(self, range_min, range_max, num) -> bool:
+    @staticmethod
+    def is_within_cq_range(range_min: Decimal, range_max: Decimal, num: ModifiedRowValue) -> bool:
         """Validation to check if a number lies within the expected range.
 
         Arguments:
             range_min {Decimal} - minimum range number
             range_max {Decimal} - maximum range number
-            num {Decimal128} - the number to be tested
+            num {ModifiedRowValue} - the number to be tested
 
         Returns:
             bool - whether the value lies within range
         """
-        # cannot compare Decimal128 to Decimal or to other Decimal128s
-        return range_min <= num.to_decimal() <= range_max
+        # TODO: this casting is dangerous and needs to be re-thought...
+        return range_min <= Decimal(cast(Decimal128, num).to_decimal()) <= range_max
```

Comment:
retain the other comment too?

Reply:
So this `cast` function is purely for the type checker? 
Is that also the reason why you've wrapped it all in `Decimal()` ?

Seems a shame to make the code less readable for the type checker. Might as well use a strongly typed language? I haven't read about it, explain it to me :)

Reply:
Yes thanks, I will add the previous helpful comment back.

I'm no expert, I'm on a journey with mypy and typing in python - it's quite new, at least to me. :P

Yes it's a shame that in this instance the readability is affected, but in general, the type annotations are helpful while coding and for catching obvious bugs. I prefer the dynamic typing of python (and python as a language) more than static typed languages like Java etc. and python's type annotations allows the benefit of types with a "scripting" language.

I have some ideas to refactor which should clean this up nicely.